src/form/input/number/NumberInput.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {useRef, RefObject} from "react";
+import classNames from "classnames";
+
+import Input, {InputProps} from "../Input";
+import {
+  NOT_INTEGER_FIRST_CHARACTER_OF_STRING_REGEX,
+  NOT_NUMBER_NOR_DECIMAL_POINT_REGEX,
+  PRECISION_REGEX,
+  IS_LAST_CHARACTER_DECIMAL_POINT_REGEX,
+  DECIMAL_NUMBER_SEPARATOR,
+  THOUSANDTHS_SEPARATOR
+} from "../../../core/utils/number/numberConstants";
+import {KEYBOARD_EVENT_KEY} from "../../../core/utils/keyboard/keyboardEventConstants";
+
+export interface NumberInputProps {
+  testid: string;
+  name: string;
+  onChange: React.ReactEventHandler<HTMLInputElement>;
+  value: string;
+  title?: string;
+  min?: number;
+  max?: number;
+  maxFractionDigits?: number;
+  placeholder?: string;
+  customClassName?: string;
+  leftIcon?: React.ReactNode;
+  onFocus?: InputProps["onFocus"];
+  onKeyDown?: InputProps["onKeyDown"];
+  prefixIconToValue?: string;
+  isDisabled?: boolean;
+  hasError?: boolean;
+}
+
+function NumberInput({
+  testid,
+  name,
+  title,
+  value,
+  min,
+  max,
+  maxFractionDigits = 0,
+  onChange,
+  hasError,
+  placeholder,
+  customClassName,
+  leftIcon,
+  onFocus,
+  onKeyDown,
+  prefixIconToValue,
+  isDisabled
+}: NumberInputProps) {
+  const className = classNames("input__number", customClassName);
+  const containerRef = useRef() as RefObject<HTMLDivElement>;
+
+  let finalValue = value;
+
+  if (finalValue && prefixIconToValue && !finalValue.includes(prefixIconToValue)) {
+    finalValue = `${prefixIconToValue} ${finalValue}`;
+  }
```

Comment:
This is not a good UX but was hack used in a project. I think we should remove this from the toolkit and have a proper UX solution for the number inputs that require displaying a symbol/unit. This require assistance from the design team so let's just remove the prop for now.